plugin/torch/src/backend_flagcx.cpp

```diff
@@ -294,6 +327,53 @@ namespace c10d
         return work;
     }
 
+    template <typename Fn>
+    c10::intrusive_ptr<Work> BackendFlagcx::collectiveCoalesced(
+        std::vector<at::Tensor>& inputs,
+        std::vector<at::Tensor>& outputs,
+        Fn fn,
+        OpType opType)
+    {
+        // Currently, the API permits one scenario where inputs.size() and
+        // outputs.size() are > 0.
+        // 1. If the call was a _coalesced call, all inputs must be on the same
+        // device.
+        //    The group of flagcx calls applies the collective separately to each input,
+        //    but the group as a whole should be efficient.
+        auto device = inputs[0].device();
+        initComm(device);
+
+        // TODO: keep track of the coalesced state at backend side.
+
+        // First let NCCL stream wait for input tensor allocation stream
+        syncStream(device);
+        auto work = c10::make_intrusive<WorkFlagcx>(opType, stream, handler->devHandle);
+
+        {
+            AutoFlagcxGroup flagcx_group_guard;
+            for (const auto i : c10::irange(inputs.size()))
+            {
+                // TODO: we need to record these input/output to prevent being freed before the collective finished.
+                auto inputTensor = inputs[i];
+                auto outputTensor = outputs[i];
+                // Perform the collective operation
+                fn(inputTensor, outputTensor, handler->comm, stream);
+            }
+        }
+
+        work->event_->record(stream, device_id);
+        work->device_id_ = device_id;
```

Comment:
We should submit another pull request to maintain consistent coding standards by changing `device_id` to `device_id_`, clearly indicating its role as a data member. This will enhance readability and uphold best practices.